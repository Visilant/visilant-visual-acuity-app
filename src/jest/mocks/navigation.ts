import { RootStackParamList } from '@screens/stack'

export const mockedNavigation = {
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  goBack: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  setParams: jest.fn(),
  dispatch: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getId: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  setOptions: jest.fn()
}

export const getMockedRoute = (routeName: keyof RootStackParamList) => ({
  key: '',
  name: routeName
})
