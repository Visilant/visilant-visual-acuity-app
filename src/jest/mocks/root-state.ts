import { RootState } from '@store'

const mockRootState: RootState = {
  _persist: { version: 1, rehydrated: false },
  examination: {
    entities: [],
    activeExamination: undefined,
    deepLinkParams: undefined
  },
  auth: {
    token: 'mock-token',
    refresh_token: 'mock-refresh-token',
    _persist: {
      version: 0,
      rehydrated: false
    }
  },
  settings: { showDashboardInstructions: true, locale: 'en-EN' },
  snackbar: { alerts: [] },
  network: {
    isQueuePaused: false,
    actionQueue: [],
    isConnected: null
  }
}

export const getMockState = () => JSON.parse(JSON.stringify(mockRootState))
