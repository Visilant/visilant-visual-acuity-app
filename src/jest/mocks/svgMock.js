export default 'svg'
export const ReactComponent = 'View'
