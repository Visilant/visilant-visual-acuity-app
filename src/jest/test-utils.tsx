import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { fireEvent, render, RenderOptions, screen } from '@testing-library/react-native'
import { WrapperProps } from './types'
import { ThemeProvider } from 'styled-components/native'
import { appTheme } from '@components/theme/theme'
import { LocalizationProvider } from '@localizations/Provider'
import { store } from '@store'

const Providers: React.FunctionComponent<WrapperProps> = ({ children }) => {
  return (
    <ReduxProvider store={store}>
      <ThemeProvider theme={appTheme}>
        <LocalizationProvider>{children}</LocalizationProvider>
      </ThemeProvider>
    </ReduxProvider>
  )
}

const customRender = <T,>(
  ui: React.ReactElement<unknown, string | React.JSXElementConstructor<T>>,
  options?: RenderOptions
) => {
  return render(ui, {
    wrapper: Providers,
    ...options
  })
}

export { fireEvent, customRender as render, screen }
