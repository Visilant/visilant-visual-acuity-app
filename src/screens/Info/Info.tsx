import { Body, Button, Card, CardRow, Header, Icon, SafeView, Screen, Text } from '@components'
import { IconsEnum } from '@components/Icon'
import { TextStyles } from '@components/Text'
import React from 'react'
import { RootStackScreenProps } from '@screens/stack'
import { withErrorBoundary } from '@hocs/withErrorBoundary'

const Info = ({ navigation }: RootStackScreenProps<'Info'>) => {
  return (
    <Screen>
      <SafeView>
        <Header>
          <Text namespace="screens" text="info.title" textStyle={TextStyles.Title} />
          <Icon icon={IconsEnum.close} action={() => navigation.navigate('Dashboard')} />
        </Header>
        <Body>
          <Card>
            <CardRow>
              <Text namespace="screens" text="info.smallTitle" textStyle={TextStyles.Label} />
            </CardRow>
            <CardRow>
              <Text namespace="screens" text="info.intro" textStyle={TextStyles.Paragraph} />
            </CardRow>
          </Card>
          <Button
            namespace="screens"
            text="info.button"
            capitalize={true}
            onPress={() => navigation.navigate('Onboarding')}
          />
        </Body>
      </SafeView>
    </Screen>
  )
}

export default withErrorBoundary(Info)
