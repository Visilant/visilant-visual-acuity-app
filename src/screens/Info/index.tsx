import { Stack } from '@screens/stack'
import React from 'react'
import Info from './Info'

const Screen = () => {
  return (
    <>
      <Stack.Screen name="Info" component={Info} options={{ title: 'Info' }} />
    </>
  )
}

export default Screen
