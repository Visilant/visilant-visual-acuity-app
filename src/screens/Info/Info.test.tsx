import { mockedNavigation } from '@jest/mocks/navigation'
import { fireEvent, render, screen } from '@jest/test-utils'
import React from 'react'
import Info from './Info'

jest.mock(
  'react-native-safe-area-context',
  () => jest.requireActual('react-native-safe-area-context/jest/mock').default
)

describe('Info screen', () => {
  test('matches the snapshot', async () => {
    const info = render(<Info navigation={mockedNavigation} route={{ key: '', name: 'Info' }} />)
    expect(info).toMatchSnapshot()
  })

  test('the "start test" button navigates away', () => {
    render(<Info navigation={mockedNavigation} route={{ key: '', name: 'Info' }} />)

    const btn = screen.getByText('ONBOARDING')
    fireEvent(btn, 'press')

    expect(mockedNavigation.navigate).toBeCalledTimes(1)
  })

  test('renders starter instructions', async () => {
    render(<Info navigation={mockedNavigation} route={{ key: '', name: 'Info' }} />)

    screen.getByText('Info & help')
    screen.getByText('How can you use it?')
    screen.getByText(
      'At Visilant, we strive to enable equitable access to eye care using telemedicine and artificial intelligence. Our intelligent telemedicine platform is designed to reach patients in the last mile and integrate them into the eye care system, so that no patient has to experience vision loss or blindness from preventable causes.'
    )
  })
})
