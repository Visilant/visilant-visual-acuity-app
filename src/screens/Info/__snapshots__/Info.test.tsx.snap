// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info screen matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#EDF1F4",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    compact={false}
    insets={
      {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 60,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomColor": "rgba(41,45,92,0.1)",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 25,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#23255A",
              "fontFamily": "Caprasimo",
              "fontSize": 36,
              "lineHeight": 38,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
            "
  ",
            [Function],
            "
  font-family: Caprasimo;
  font-size: 36px;
  line-height: 38px;
",
          ]
        }
      >
        Info & help
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 25,
            "marginRight": 0,
            "width": 25,
          }
        }
      >
        <svg
          fill="#23255A"
        />
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "flexDirection": "column",
          "flexGrow": 0,
          "gap": 35,
          "justifyContent": "flex-start",
          "paddingBottom": 35,
          "paddingTop": 30,
        }
      }
      style={
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View>
        <View
          style={
            {
              "backgroundColor": "rgba(255,255,255,0.5)",
              "borderBottomLeftRadius": 25,
              "borderBottomRightRadius": 25,
              "borderColor": "#e6eef4",
              "borderStyle": "solid",
              "borderTopLeftRadius": 25,
              "borderTopRightRadius": 25,
              "borderWidth": 1,
              "gap": 30,
              "paddingBottom": 30,
              "paddingLeft": 25,
              "paddingRight": 25,
              "paddingTop": 25,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "columnGap": 15,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "rowGap": 20,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#23255A",
                    "fontFamily": "Libre-Franklin-SemiBold",
                    "fontSize": 20,
                    "lineHeight": 26,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                  "
  ",
                  [Function],
                  "
  font-size: 20px;
  font-family: Libre-Franklin-SemiBold;
  line-height: 26px;
",
                ]
              }
            >
              How can you use it?
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "columnGap": 15,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "rowGap": 20,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#23255A",
                    "fontFamily": "Libre-Franklin-ExtraLight",
                    "fontSize": 18,
                    "lineHeight": 28,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                  "
  ",
                  [Function],
                  "
  font-size: 18px;
  font-family: Libre-Franklin-ExtraLight;
  line-height: 28px;
",
                ]
              }
            >
              At Visilant, we strive to enable equitable access to eye care using telemedicine and artificial intelligence. Our intelligent telemedicine platform is designed to reach patients in the last mile and integrate them into the eye care system, so that no patient has to experience vision loss or blindness from preventable causes.
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          size={1}
          style={
            {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#EF4823",
              "borderBottomLeftRadius": 31,
              "borderBottomRightRadius": 31,
              "borderTopLeftRadius": 31,
              "borderTopRightRadius": 31,
              "flexWrap": "nowrap",
              "height": 62,
              "justifyContent": "center",
              "overflow": "hidden",
              "paddingBottom": 12,
              "paddingLeft": 30,
              "paddingRight": 30,
              "paddingTop": 12,
            }
          }
          type="normal"
        >
          <View
            style={{}}
          />
          <Text
            props={
              {
                "size": 1,
                "type": "normal",
              }
            }
            style={
              [
                {
                  "color": "#E6EEF4",
                  "flexWrap": "nowrap",
                  "fontFamily": "Libre-Franklin-Bold",
                  "fontSize": 24,
                  "lineHeight": 38,
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "top": -1,
                },
                "
  position: relative;
  top: ",
                [Function],
                ";
  color: ",
                [Function],
                ";
  font-family: Libre-Franklin-Bold;

  ",
                [Function],
                ";
  text-align: center;
  flex-wrap: nowrap;
",
              ]
            }
          >
            ONBOARDING
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
